src/Lykke.AzureStorage/Blob/AzureBlob.cs

```diff
@@ -103,17 +103,25 @@ private BlobRequestOptions GetRequestOptions()
             return containerRef.GetBlockBlobReference(key);
         }
 
-        public async Task SaveBlobAsync(string container, string key, byte[] blob)
+        public async Task SaveBlobAsync(string container, string key, byte[] blob, IReadOnlyDictionary<string, string> metadata = null)
         {
             var blockBlob = await GetBlockBlobReferenceAsync(container, key, createIfNotExists: true);
 
             var mimeType = ContentTypesHelper.GetContentType(key);
-
+            
             if (mimeType != null)
             {
                 blockBlob.Properties.ContentType = mimeType;
             }
 
+            if (metadata != null)
+            {
+                foreach (var keyValuePair in metadata)
+                {
+                    blockBlob.Metadata[keyValuePair.Key] = keyValuePair.Value;
+                }
+            }
```

Comment:
You need to call SetMetadataAsync here to upload metadata changes

Reply:
blockBlob.UploadFromByteArrayAsync on the next lines is doing the trick and uploading them together with blockBlob.Properties.